lib/Jit/jitoptions.cpp

```diff
@@ -50,11 +58,52 @@ void freeStringConfigValue(ICorJitInfo *CorInfo, UTF16CharT *Value) {
 }
 
 JitOptions::JitOptions(LLILCJitContext &Context) {
-  // Set 'IsAltJit' based on environment information.
-  IsAltJit = queryIsAltJit(Context);
+
+  if (!StaticOptionsInitialized) {
+    StaticDumpLevel = ::DumpLevel::NODUMP;
+
+    char16_t *LevelWStr =
+      getStringConfigValue(Context.JitInfo, UTF16("DUMPLLVMIR"));
+    if (LevelWStr) {
+      std::unique_ptr<std::string> Level = Convert::utf16ToUtf8(LevelWStr);
+      std::transform(Level->begin(), Level->end(), Level->begin(), ::toupper);
+      if (Level->compare("VERBOSE") == 0) {
+        StaticDumpLevel = ::DumpLevel::VERBOSE;
+
+      } else if (Level->compare("SUMMARY") == 0) {
+        StaticDumpLevel = ::DumpLevel::SUMMARY;
+      }
+    }
+    if (Context.Flags & CORJIT_FLG_PREJIT) {
+      InitMethodSet(Context, AltJitNgenMethodSet, UTF16("AltJitNgen"));
+      // Set up AltJitNgen set to be used for AltJit test.
+      AltJitPtr = &AltJitNgenMethodSet;
+    } else {
+      InitMethodSet(Context, AltJitMethodSet, UTF16("AltJit"));
+      // Set up AltJit set to be use for AltJit test.
+      AltJitPtr = &AltJitMethodSet;
+    }
+    InitMethodSet(Context, ExcludeMethodSet, UTF16("AltJitExclude"));
+    InitMethodSet(Context, BreakMethodSet, UTF16("AltJitBreak"));
+    InitMethodSet(Context, MSILMethodSet, UTF16("AltJitMSILDump"));
+    InitMethodSet(Context, LLVMMethodSet, UTF16("AltJitLLVMDump"));
+    InitMethodSet(Context, EntrySizeMethodSet, UTF16("AltJitCodeRangeDump"));
+    StaticOptionsInitialized = true;
+  }
+
 
   // Set dump level for this JIT invocation.
-  DumpLevel = queryDumpLevel(Context);
+  DumpLevel = StaticDumpLevel;
+  const char *ClassName = nullptr;
+  const char *MethodName = 
+    Context.JitInfo->getMethodName(Context.MethodInfo->ftn, &ClassName);
+  PCCOR_SIGNATURE PCSig = Context.MethodInfo->args.pSig;
+  IsAltJit = AltJitPtr->contains(MethodName, ClassName, PCSig);
```

Comment:
IsAltJit was already hoisted out into it's own initializer?  Why did this change?